uv/reporter/state.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/python
+#
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Reporter interface and implementations."""
+
+from contextlib import contextmanager
+from typing import Dict
+
+import mlflow as mlf
+import uv.types as t
+import uv.util.env as ue
+from uv.reporter.base import AbstractReporter
+from uv.reporter.store import NullReporter
+
+_active_reporter = None
```

Comment:
now the user can set a global reporter, and call `uv.report` and `uv.report_all` anywhere. I don't like... global... state... but this is more "PYTHONIC!" and you can still use the old way.